### PR TITLE
Docs for new [[build.env]] table in builder.toml

### PR DESCRIPTION
**Pull request by WYGIN**
this will add docs for the pr [buildpacks/pack#1926](https://github.com/buildpacks/pack/pull/1926)

### CODE REVIEW

File: content/docs/reference/config/builder-config.md [hunk 1]
```diff
@@ -73,6 +73,15 @@ A [builder][builder] configuration schema is as follows:
     > at the time of `pack`'s release. In other words, for a particular version of `pack`, this default
     > will not change despite new lifecycle versions being released.
 
+- #### `build.env` _(optional)_
+  The [[build.env]] is used to override platform environment variables at build-time. use Env `CNB_BUILD_CONFIG_DIR` to override when building image from cli to change default(`/cnb/build-config`) directory of these build-time environment variables stored
```
Comment (natalieparellano) [inline content/docs/reference/config/builder-config.md:77]:
```suggestion
  The [[build.env]] is used to specify [operator-defined](https://github.com/buildpacks/spec/blob/main/platform.md#operator-defined-variables) build-time environment variables for buildpacks. Set `CNB_BUILD_CONFIG_DIR` in pack's environment to override the default directory (/cnb/build-config/env) where environment variable files are saved within the builder.
```

File: content/docs/reference/config/builder-config.md [hunk 2]
```diff
@@ -73,6 +73,15 @@ A [builder][builder] configuration schema is as follows:
...
+
+  - **`name`** _(string, required)_\
+    The Name/Key of the Environment Variable. If Platform Environment Variable key/name is specified, then it is overridden at build-time else it will create a build-time environment variable with the given `name` that is not defined(key/name doesn't exists) at runtime of the container
+
+  - **`value`** _(string, required)_\
```
Comment (natalieparellano) [inline content/docs/reference/config/builder-config.md:82]: Should we mention suffix and delim here? We can link to https://github.com/buildpacks/spec/blob/main/buildpack.md#environment-variable-modification-rules to make the text here more terse if we want to.

**Review by AidanDelaney — APPROVED** [review]
This is *great*.  The build failures are not a consequence of this change.  I'll get those fixed and then we can merge this.

File: content/docs/reference/config/builder-config.md [hunk 3]
```diff
@@ -73,6 +73,23 @@ A [builder][builder] configuration schema is as follows:
     > at the time of `pack`'s release. In other words, for a particular version of `pack`, this default
     > will not change despite new lifecycle versions being released.
 
+- #### `build.env` _(optional)_
+  The [[[build.env]]](https://github.com/buildpacks/spec/blob/main/buildpack.md#environment-variable-modification-rules) is used to specify [operator-defined](https://github.com/buildpacks/spec/blob/main/platform.md#operator-defined-variables) build-time environment variables for buildpacks. Set `CNB_BUILD_CONFIG_DIR` in pack's environment to override the default directory (`/cnb/build-config/env`) where environment variable files are stored within the builder.
+
+  - **`name`** _(string, required)_\
+    The name/key of the environment variable. If a platform environment variable with the given key/name exists, it will be overridden at build time. Otherwise, a new build-time environment variable with the given name will be created.
```
Comment (natalieparellano) [inline content/docs/reference/config/builder-config.md:80]:
```suggestion
    The name/key of the environment variable.
```

File: content/docs/reference/config/builder-config.md [hunk 4]
```diff
@@ -73,6 +73,23 @@ A [builder][builder] configuration schema is as follows:
...
+
+  - **`value`** _(string, required)_\
+    The value of the specified environment variable, depends on the `suffix`.
```
Comment (natalieparellano) [inline content/docs/reference/config/builder-config.md:83]:
```suggestion
    The value of the environment variable.
```

File: content/docs/reference/config/builder-config.md [hunk 5]
```diff
@@ -73,6 +73,23 @@ A [builder][builder] configuration schema is as follows:
...
+
+  - **`suffix`** _(string, optional)_\
+    The type of action performed on platform environment variables, one of [`default`](https://github.com/buildpacks/spec/blob/main/buildpack.md#default), [`override`](https://github.com/buildpacks/spec/blob/main/buildpack.md#override), [`append`](https://github.com/buildpacks/spec/blob/main/buildpack.md#append), or [`prepend`](https://github.com/buildpacks/spec/blob/main/buildpack.md#prepend). Defaults to `default` if this field is omited.
```
Comment (natalieparellano) [inline content/docs/reference/config/builder-config.md:86]:
```suggestion
    The type of action used to modify the environment variable when end-users or buildpacks define the same name/key, one of [`default`](https://github.com/buildpacks/spec/blob/main/buildpack.md#default), [`override`](https://github.com/buildpacks/spec/blob/main/buildpack.md#override), [`append`](https://github.com/buildpacks/spec/blob/main/buildpack.md#append), or [`prepend`](https://github.com/buildpacks/spec/blob/main/buildpack.md#prepend). Defaults to `default` if this field is omitted. Operator-defined environment variables take precedence over end-user or buildpack-defined environment variables.
```

**Review by natalieparellano — COMMENTED** [review]
This is awesome :) I made a few suggestions for readability.

**Review by natalieparellano — APPROVED** [review]

**Pull request merged**